matron/src/hardware/platform.c

```diff
@@ -0,0 +1,22 @@
+#include <unistd.h>
+#include <stdlib.h>
+
+#include "platform.h"
+
+platform_t p;
+
+void init_platform() {
+  if(access("/sys/firmware/devicetree/base/model", F_OK ) != -1) {
+    if(system("sudo cat /sys/firmware/devicetree/base/model | grep 'Compute'")) {
```

Comment:
Is `sudo` needed here / potentially problematic? The `access` call should have already confirmed that you have read access to the file.

Reply:
ah it's probably unnecessary, i was doing some unconscious cut-paste. will test/fix